libs/hdf-converters/src/utils/attestations.ts

```diff
@@ -50,11 +50,29 @@ export function advanceDate(
       date.add(1, 'day');
       break;
     default:
-      const msTime = ms(frequency);
-      if (!msTime) {
-        throw new Error('Unknown date format: ' + frequency);
-      } else {
-        date.add(msTime, 'milliseconds');
+      const match = frequency.match(/(\d+(?:\.\d+)?)([a-z]+)/);
+      
+      // throw error if input doesn't match specified format of <x>d/wk/m/yr
+      if (!match){ throw new Error('Unknown date format: '+frequency); };
+
+      const freq_num = match[1];
+      const unit = match[2];
+
+      switch (match[2]) {
```

Comment:
this switch is unnecessary and in fact will lead to us throwing an error even when the user puts in something like the word 'days' themselves.  imo let's just feed whatever we receive as input from the user into the library and catch -> throw errors from that library but with a nicer error statement probably.

also now that we're throwing this error, we should probably have good error handling wherever this function is called.